### PR TITLE
Merge deployment scripts into one

### LINKED ISSUE #532: Have only one deployment script
Resulting from the big hyperion merge, there are currently two separate scripts to deploy mx-bluesky on beamlines: one is the old mx-bluesky one and one is for deploying hyperion. This can be slightly confusing and we should have only one script.
The two scripts are fairly similar, the main thing to be aware of is that for hyperion we're installing on the control machine, which is not always what we want, so a flag might be added for this.

## Acceptance Criteria
- There is only one deployment script which installs mx-bluesky correctly on any beamline.
- The docs relative to deployment have been updated.

### PR DESCRIPTION
**Pull request by noemifrisina**
One script to deploy them all. Closes #532 

To test:
- Run test deployment and check that it complete without errors.
- Check docs are make sense.

### CODE REVIEW

File: utility_scripts/deploy/create_venv.py [hunk 1]
```diff
@@ -3,21 +3,23 @@
...
 
-def setup_venv(path_to_create_venv_script, deployment_directory):
-    # Set up environment and run /dls_dev_env.sh...
-    os.chdir(deployment_directory)
-    print(f"Setting up environment in {deployment_directory}")
-
-    with Popen(
-        path_to_create_venv_script, stdout=PIPE, bufsize=1, universal_newlines=True
-    ) as p:
+def run_process_and_print_output(proc_to_run):
```
Comment (rtuck99) [inline utility_scripts/deploy/create_venv.py:6]: We should probably capture stderr here as well and print it

File: utility_scripts/deploy/deploy_mx_bluesky.py [hunk 2]
```diff
@@ -204,38 +210,99 @@ def create_symlink_by_tmp_and_rename(dirname, target, linkname):
...
+    parser.add_argument(
+        "--print-release-dir",
+        action="store_true",
+        help="Print the path to the release folder and then exit.",
+    )
+    parser.add_argument(
+        "-nc",
+        "--no-control",
+        action="store_false",
+        help="Create environment from the control machine.",
```
Comment (rtuck99) [inline utility_scripts/deploy/deploy_mx_bluesky.py:287]: This is confusingly worded! Can it be changed to something like "If running from the control machine use this to create the venv directly and not via ssh"

File: utility_scripts/deploy/deploy_mx_bluesky.py [hunk 3]
```diff
@@ -139,22 +136,24 @@ def _create_environment_from_control_machine(
...
-        name="hyperion", repo_args=os.path.join(this_repo_top, ".git"), options=options
+    mx_repo = Deployment(
+        name="mx-bluesky",
+        repo_args=os.path.join(this_repo_top, ".git"),
+        options=options,
     )
 
-    if hyperion_repo.name != "hyperion":
-        raise ValueError("This function should only be used with the hyperion repo")
+    if mx_repo.name != "mx-bluesky":
```
Comment (rtuck99) [inline utility_scripts/deploy/deploy_mx_bluesky.py:152]: why would the repo name ever not be mx-bluesky given we just constructed it?

File: utility_scripts/deploy/deploy_mx_bluesky.py [hunk 4]
```diff
@@ -170,30 +169,37 @@ def main(options: Options):
...
+                    mx_repo, path_to_create_venv, path_to_dls_dev_env, beamline
                 )
             else:
-                setup_venv(path_to_create_venv, hyperion_repo.deploy_location)
+                setup_venv(path_to_dls_dev_env, mx_repo.deploy_location)
+
+    # If on beamline I24 also deploy the screens to run ssx collections
+    if beamline == "i24":
+        print("Setting up edm screens for serial collections on I24.")
+        run_process_and_print_output("./utility_scripts/deploy/deploy_edm_for_ssx.sh")
```
Comment (rtuck99) [inline utility_scripts/deploy/deploy_mx_bluesky.py:203]: Can this be a top-level function please

**Comment by rtuck99** [comment]
I was going to add lots of comments about the documentation but instead I raised a separate GH issue #578

File: docs/developer/general/how-to/deploy-a-release.rst [hunk 5]
```diff
@@ -3,68 +3,41 @@ Deploy a New Release
...
 
+If you want to test the script for a specific beamline you can run:
 
 .. code:: console
 
-    python ./deploy/deploy_mxbluesky.py --beamline i24 --dev-path /your-path/
-
-which will create the beamline deployment (eg. I24) in the specified test directory ``/your-path/mxbluesky_release_test``.
-
+    python ./deploy/deploy_mx_bluesky.py i03 --dev
```
Comment (rtuck99) [inline docs/developer/general/how-to/deploy-a-release.rst:18]: `utility_scripts/deploy/deploy_mx_bluesky.py`

File: docs/developer/general/how-to/deploy-a-release.rst [hunk 6]
```diff
@@ -3,68 +3,41 @@ Deploy a New Release
...
 
-**Note:** When deploying on I24, the edm screens for serial crystallography will be deployed automatically along with the mx-bluesky release. 
-When running a ``dev`` deployment instead, `this script <https://github.com/DiamondLightSource/mx-bluesky/wiki/Serial-Crystallography-on-I24#deploying-a-local-version-of-the-edm-screens>`_ will also need to be run to get the latest version of the screens.
 
+which will create the beamline deployment of the new release in ``/scratch/30day_tmp/mxbluesky_release_test``.
```
Comment (rtuck99) [inline docs/developer/general/how-to/deploy-a-release.rst:21]: mx-bluesky_release_test

**Review by rtuck99 — APPROVED** [review]
Approved with above minor suggested changes

File: utility_scripts/deploy/deploy_mx_bluesky.py (hunk 2 above)
Comment (noemifrisina) [reply utility_scripts/deploy/deploy_mx_bluesky.py:287]: Actually, I think I just forgot to change the help message when I changed the code. This is an option to NOT use the control machine for deployment, so will re-write this

**Pull request merged**